packages/inter-protocol/src/proposals/replaceElectorate.js

```diff
@@ -214,6 +227,23 @@ const startNewEconomicCommittee = async (
   trace(`committeeName ${committeeName}`);
   trace(`committeeSize ${committeeSize}`);
 
+  // if retiredContractInstances doesn't exist, create it.
+  const contractInstanceMap = makeScalarBigMapStore(
+    'retiredContractInstances',
+    { durable: true },
+  );
+  retiredInstanceWriter.resolve(contractInstanceMap);
```

Comment:
please extract a function that documents why it can be called in multiple places at any time.

it should also document the naming scheme: {agoricNames.instance key}-{boardID}

Reply:
The most natural form is generic creation of a Map in promiseSpace, but this doesn't leave a place to document the usage of retiredContractInstances. Would you rather have a shared function called createRetiredContractInstancesMap, which could document the pattern?

Reply:
[f4c67ec](https://github.com/Agoric/agoric-sdk/pull/10680/commits/f4c67ec6be2d0b0be8a0beb630f7c18c3617c8ca) is okay but consider something more direct: https://github.com/Agoric/agoric-sdk/commit/515f772cdb5fffe0a20763450f65fc0f6531b886